tidy3d/components/geometry/base.py

```diff
@@ -2056,10 +2544,52 @@ def to_polygon_list(base_geometry: Shapely) -> List[Shapely]:
             return [base_geometry]
         return []
 
+    def intersections_tilted_plane(
+        self, normal: Coordinate, origin: Coordinate, to_2D: MatrixReal4x4
+    ) -> List[Shapely]:
+        """Return a list of shapely geometries at the plane specified by normal and origin.
+
+        Parameters
+        ----------
+        normal : Coordinate
+            Vector defining the normal direction to the plane.
+        origin : Coordinate
+            Vector defining the plane origin.
+        to_2D : MatrixReal4x4
+            Transformation matrix to apply to resulting shapes.
+
+        Returns
+        -------
+        List[shapely.geometry.base.BaseGeometry]
+            List of 2D shapes that intersect plane.
+            For more details refer to
+            `Shapely's Documentaton <https://shapely.readthedocs.io/en/stable/project.html>`_.
+        """
+        geom_a = Geometry.evaluate_inf_shape(
+            shapely.unary_union(self.geometry_a.intersections_tilted_plane(normal, origin, to_2D))
+        )
+        geom_b = Geometry.evaluate_inf_shape(
+            shapely.unary_union(self.geometry_b.intersections_tilted_plane(normal, origin, to_2D))
+        )
+        if self.operation == "union":
```

Comment:
This is just organizational, but it might help to have a dictionary somewhere defining the mapping of `ClipOperation.operation` to the shapely operation.

```py
operation_mapping = dict(
    union=shapely.union,
    intersection=shapely.intersection,
    difference=shapely.difference,
    symmetric_difference=shapely.symmetric_difference,
)
```
and then this block of code beomes
```py
operation_fn = operation_mapping[self.operation]
operation_result = operation_fn(geom_a, geom_b)
return ClipOperation.to_polygon_list(operation_result)
```

maybe this mapping could be useful in other places as well.

Reply:
I've also added an equivalent for bit-type operations.